frontend/src/metabase/visualizations/lib/settings/graph.js

```diff
@@ -151,26 +151,38 @@ export const GRAPH_DATA_SETTINGS = {
     dashboard: false,
     useRawSeries: true,
   },
+  "graph.series_order_dimension": {
+    getValue: (_series, settings) => settings["graph.dimensions"][1],
+    readDependencies: ["graph.series_order"],
+  },
   "graph.series_order": {
     section: t`Data`,
     widget: ChartSettingOrderedSimple,
     marginBottom: "1rem",
-    isValid: (series, settings) => {
+
+    getValue: (series, settings) => {
```

Comment:
If `getValue` is defined, we will always use this function to determine the value and functions like `isValid` and `getDefault` will not be run.